### PR TITLE
chore: sync bgfx and bump ios-cmake

### PR DESCRIPTION
**Pull request by okwasniewski**
This PR syncs bgfx and bumps ios-cmake (contains necessary fixes for visionOS work).

### CODE REVIEW

**Comment by okwasniewski** [comment]
Hey @CedricGuillemet @ryantrem,

Can you help me out with the Windows build failing? Looks like its something related to `arcana` library after the update.

CleanShot 2024-08-22 at 17 10 21@2x

**Comment by CedricGuillemet** [comment]
I'm taking a look

File: CMakeLists.txt [hunk 1]
```diff
@@ -141,6 +141,9 @@ if(MSVC)
     # https://gitlab.kitware.com/cmake/cmake/-/issues/18837
     add_compile_options(/Zc:__cplusplus)
 
+    # https://devblogs.microsoft.com/cppblog/announcing-full-support-for-a-c-c-conformant-preprocessor-in-msvc/
```
Comment (CedricGuillemet) [inline CMakeLists.txt:144]: @okwasniewski why adding this option?

File: CMakeLists.txt [hunk 2]
```diff
@@ -34,7 +34,7 @@ FetchContent_Declare(googletest
     URL "https://github.com/google/googletest/archive/refs/tags/v1.14.0.tar.gz")
 FetchContent_Declare(ios-cmake
     GIT_REPOSITORY https://github.com/leetal/ios-cmake.git
-    GIT_TAG 4.4.1)
+    GIT_TAG 4.5.0)
```
[the remark this answers is not in the document]
Comment (ryantrem) [reply CMakeLists.txt:37]: Oh nm, I see your comment on the other PR about wanting to test this out.

[the remark this answers is not in the document]
Comment (okwasniewski) [reply CMakeLists.txt:37]: No, it's not required just wanted to decouple this from my visionOS PR. I can create a separate just bumping iOS-Cmake

[the remark this answers is not in the document]
Comment (ryantrem) [reply CMakeLists.txt:37]: I think it's ok to include it here if it doesn't require a bunch of other changes.

File: CMakeLists.txt (hunk 1 above)
Comment (okwasniewski) [reply CMakeLists.txt:144]: Without this the build was failing with a message to add it, here is a discussion about this flag: https://github.com/BabylonJS/bgfx/pull/39#issuecomment-2273805547

Comment (CedricGuillemet) [reply CMakeLists.txt:144]: I guess arcana.cpp is not conformat with the wstring macro. let me see if I can find a fix...

Comment (CedricGuillemet) [reply CMakeLists.txt:144]: build is fine with conformant macro : https://github.com/BabylonJS/BabylonNative/pull/1409
I've done the PR for arcana.cpp https://github.com/microsoft/arcana.cpp/pull/45 @ryantrem 
When the PR is merged, you can update the commit id in this PR @okwasniewski

File: Apps/Playground/CMakeLists.txt [hunk 3]
```diff
@@ -173,6 +174,8 @@ if(APPLE)
             FOLDER "Playground")
     endif()
     set_property(TARGET Playground PROPERTY UNITY_BUILD false)
+    set_property(TARGET Playground PROPERTY XCODE_GENERATE_SCHEME YES)
+    set_property(TARGET Playground PROPERTY XCODE_ATTRIBUTE_CODE_SIGNING_ALLOWED YES)
```
Comment (okwasniewski) [inline Apps/Playground/CMakeLists.txt:178]: This change is an additional fix required by the bump of `ios-cmake`. 

More details: https://github.com/leetal/ios-cmake/pull/208#issuecomment-2235947477

File: Apps/Playground/CMakeLists.txt [hunk 4]
```diff
@@ -173,6 +174,8 @@ if(APPLE)
             FOLDER "Playground")
     endif()
     set_property(TARGET Playground PROPERTY UNITY_BUILD false)
+    set_property(TARGET Playground PROPERTY XCODE_GENERATE_SCHEME YES)
```
Comment (okwasniewski) [inline Apps/Playground/CMakeLists.txt:177]: This property `XCODE_GENERATE_SCHEME` removes unnecessary schemes from Xcode.

Now Xcode shows only necessary schemes as valid run options:

CleanShot 2024-08-23 at 15 47 32@2x

File: CMakeLists.txt (hunk 1 above)
Comment (CedricGuillemet) [reply CMakeLists.txt:144]: arcana.cpp change is merged

Comment (okwasniewski) [reply CMakeLists.txt:144]: @CedricGuillemet thanks for letting me know. Looks like the build is now green!

I think we can merge the PR: 

- https://github.com/BabylonJS/bgfx/pull/40
- Then: https://github.com/BabylonJS/bgfx.cmake/pull/76

And I will update this one with correct git repos.

Comment (ryantrem) [reply CMakeLists.txt:144]: @okwasniewski the bgfx one is merged, please go ahead and update the bgfx.cmake one with commit 207e246af754bfdd0e46229997e4c18dbafa6c04.

File: CMakeLists.txt [hunk 5]
```diff
@@ -14,16 +14,16 @@ FetchContent_Declare(AndroidExtensions
...
 FetchContent_Declare(arcore-android-sdk
     GIT_REPOSITORY https://github.com/google-ar/arcore-android-sdk.git
     GIT_TAG 98cb803de5482fb2b36effe8be3b5a0d3b726976)
 FetchContent_Declare(base-n
     GIT_REPOSITORY https://github.com/azawadzki/base-n.git
     GIT_TAG 7573e77c0b9b0e8a5fb63d96dbde212c921993b4)
 FetchContent_Declare(bgfx.cmake
-    GIT_REPOSITORY https://github.com/BabylonJS/bgfx.cmake.git
-    GIT_TAG 345e3e28a9912983fc64d1dcb4da22a445afe3fd)
+    GIT_REPOSITORY https://github.com/okwasniewski/bgfx.cmake.git
```
Comment (CedricGuillemet) [inline CMakeLists.txt:25]: Just need this commit to be updated and it's good to go :)

Comment (okwasniewski) [reply CMakeLists.txt:25]: @CedricGuillemet we also need this one merged, right? https://github.com/BabylonJS/bgfx.cmake/pull/76

I've just pointed it at the last merged commit

Comment (CedricGuillemet) [reply CMakeLists.txt:25]: I've juste merged it.

Comment (okwasniewski) [reply CMakeLists.txt]: Great, I've changed the CMakeLists to point at upstream repo

**Comment by okwasniewski** [comment]
@CedricGuillemet Looks like the build is green

**Review by ryantrem — APPROVED** [review]

**Pull request merged**